src/pages/workspaces/workspace/analysis/modals/ExportAnalysisModal/useAnalysisExportState.ts

```diff
@@ -0,0 +1,86 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { useWorkspaces } from 'src/components/workspace-utils'
+import { AnalysisProvider } from 'src/libs/ajax/analysis-providers/AnalysisProvider'
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData'
+import { useMetricsEvent } from 'src/libs/ajax/metrics/useMetrics'
+import Events, { extractCrossWorkspaceDetails } from 'src/libs/events'
+import { useCancellation } from 'src/libs/react-utils'
+import LoadedState from 'src/libs/type-utils/LoadedState'
+import { WorkspaceInfo, WorkspaceWrapper } from 'src/libs/workspace-utils'
+import { AnalysisFile } from 'src/pages/workspaces/workspace/analysis/useAnalysisFiles'
+import { getDisplayName } from 'src/pages/workspaces/workspace/analysis/utils/file-utils'
+import { ToolLabel } from 'src/pages/workspaces/workspace/analysis/utils/tool-utils'
+
+
+export type LoadedAnalysisNames = LoadedState<string[], unknown>
+
+export interface AnalysisExportState {
+  workspaces: WorkspaceWrapper[]
+  selectWorkspace: (workspaceId: string) => void
+  selectedWorkspace: WorkspaceInfo | null
+  existingAnalysisNames: LoadedAnalysisNames
+  copyAnalysis: (newName: string) => void
+  pendingCopy: LoadedState<true, unknown>
+}
+
+export const errors = {
+  badWorkspace: 'Selected Workspace does not exist',
+  noWorkspace: 'No workspace selected'
+}
+
+export const useAnalysesExportState = (sourceWorkspace: WorkspaceWrapper, printName: string, toolLabel: ToolLabel): AnalysisExportState => {
+  const { captureEvent } = useMetricsEvent()
+  const signal = useCancellation()
+  const workspaces: WorkspaceWrapper[] = useWorkspaces().workspaces
+  const [selectedWorkspace, setSelectedWorkspace] = useState<WorkspaceWrapper | null>(null)
+  const [existingAnalysisNames, setExistingAnalysisNames] = useLoadedData<string[]>()
+  const [pendingCopy, setPendingCopy] = useLoadedData<true>()
+
+  const doSelectWorkspace = async (workspaceId: string): Promise<void> => {
+    await setExistingAnalysisNames(async () => {
+      const foundWorkspaceWrapper = _.find({ workspace: { workspaceId } }, workspaces)
+      if (foundWorkspaceWrapper === undefined) {
+        throw (Error(errors.badWorkspace))
+      }
+      const chosenWorkspace = foundWorkspaceWrapper.workspace
+      setSelectedWorkspace(foundWorkspaceWrapper)
+
+      const selectedAnalyses: AnalysisFile[] = await AnalysisProvider.listAnalyses(chosenWorkspace, signal)
```

Comment:
Hmm... I like the idea of returning this `selectedAnalyses` directly, so it's clearer this statehook returns a list of files, and let the component transform the data with the map on line 50.  Seems like `doSelectWorkspace` has a bit too many responsibilities currently, and that having its responsibilities end after the provider interaction could alleviate this.